src/Tests/VCEL.Test/ToJSCodeTests.cs

```diff
@@ -16,18 +17,22 @@ public ToJsCodeTests()
         }
 
         [TestCase("fd.startsWith('AAA')==false and P < Prev(P)", "(((vcelContext.fd ? vcelContext.fd.startsWith('AAA') : false) === false) && (vcelContext.P < (Prev(vcelContext.P))))")]
-        [TestCase("t == 'C'", "(vcelContext.t === 'C')")]
+        [TestCase("t == 'C'", "((vcelContext.t === null || vcelContext.t === void 0 ? void 0 : vcelContext.t.valueOf()) === 'C')")]
         [TestCase("(D > 500000 or D < -500000)", "((vcelContext.D > 500000) || (vcelContext.D < -500000))")]
         [TestCase("code matches '(?:.+,|^)([0-9]\\d\\d)(?:,.+|$)'", "new RegExp(/(?:.+,|^)([0-9]\\d\\d)(?:,.+|$)/gm).test(vcelContext.code)")]
-        [TestCase("(K == 'C' or K == 'AC')", "((vcelContext.K === 'C') || (vcelContext.K === 'AC'))")]
+        [TestCase("(K == 'C' or K == 'AC')", "(((vcelContext.K === null || vcelContext.K === void 0 ? void 0 : vcelContext.K.valueOf()) === 'C') || ((vcelContext.K === null || vcelContext.K === void 0 ? void 0 : vcelContext.K.valueOf()) === 'AC'))")]
         [TestCase("(a < t and a > 0)", "((vcelContext.a < vcelContext.t) && (vcelContext.a > 0))")]
-        [TestCase("s == 'ACTIVE'", "(vcelContext.s === 'ACTIVE')")]
+        [TestCase("s == 'ACTIVE'", "((vcelContext.s === null || vcelContext.s === void 0 ? void 0 : vcelContext.s.valueOf()) === 'ACTIVE')")]
         public void TestJsParser_ProdRulesExamples(string expr, string expected)
```

Comment:
It would be useful to add a test to compare two variables such as
a == b
also a test case where variable is the right value:
'active' == s

Reply:
Added